samples/drivers/uart/single_wire/sample.yaml

```diff
@@ -0,0 +1,6 @@
+sample:
+  name: STM32 Single Wire UART sample
+tests:
+  sample.drivers.uart.single_wire:
+    platform_allow: stm32f3_disco
+    tags: drivers uart
```

Comment:
Would be nice to add a harness to check the sample is behaving correctly in CI.
Cf hello world sample.

Reply:
yeah, I was also thinking about that, but did not figure out, how to do that. Already asked on discord https://discord.com/channels/720317445772017664/733037944922964069/925822535533223957, but did not get an answer so far. Will have a look at hello world sample

Reply:
Do I understand correctly from the question at discord, that this test requires an STM board with a physical connection between certain pins? Upstream CI is only able of executing tests/samples on emulated/simulated platforms and only build for physical boards. As for harnesses: Most samples use console harness (as hello_word) where the console output is verified according to what is in a yaml. There is also ztest harness allowing to use asserts in a code. 
If the physical connection between the pins is required you need to add "fixture" field. E.g https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/uart/uart_async_api/testcase.yaml

Reply:
I have added the fixture and described its usage in the README.